routetable/route_table.go

```diff
@@ -341,27 +341,41 @@ func (r *RouteTable) syncRoutesForLink(ifaceName string) error {
 	if updatesFailed {
 		// Recheck whether the interface exists so we don't produce spammy logs during
 		// interface removal.
-		return r.filterErrorByIfaceState(ifaceName, UpdateFailed)
+		return r.filterErrorByIfaceState(ifaceName, UpdateFailed, UpdateFailed)
 	}
 
 	return nil
 }
 
-// filterErrorByIfaceState checks the current state of the interface; it's down or gone, it returns
-// IfaceDown or IfaceError, otherwise, it returns the given defaultErr.
-func (r *RouteTable) filterErrorByIfaceState(ifaceName string, defaultErr error) error {
-	logCxt := r.logCxt.WithField("ifaceName", ifaceName)
+// filterErrorByIfaceState checks the current state of the interface; if it's down or gone, it
+// returns IfaceDown or IfaceNotPresent, otherwise, it returns the given defaultErr.
+func (r *RouteTable) filterErrorByIfaceState(ifaceName string, currentErr, defaultErr error) error {
+	logCxt := r.logCxt.WithFields(log.Fields{"ifaceName": ifaceName, "error": currentErr})
```

Comment:
Does adding the error as a field get treated differently from using WithError() ?